### DISCUSSION TITLE
Can i use Netsparkle updater asynchronously in a seperate Process ?

### DISCUSSION THREAD

**Discussion by balaji222528** [NetSparkleUpdater/NetSparkle#591] [General]
Imagine i  have a main MAUI app running with UI etc and in case i  need to force update the current process, can i spin up another UI process that runs the netsparkle updater ? will it work cross platform or only on windows ?

**Comment by Deadpikle** [comment]
@balaji222528 

I don't see why not. You may need to manually configure the `Configuration` and `AssemblyAccessor` objects for your use case and to make sure it reads/knows version information properly, but it should work. You may also be interested in the `PreparingToExitAsync` or `PreparingToExit` events, as you'll have to kill the main software in order to update it (again, depends on your use case). By default, the process that runs NetSparkle is the one replaced / killed during the update process, but you can configure things and override functions as needed to get the behavior you want.

> will it work cross platform or only on windows ?

`NetSparkle.UI.Avalonia` is cross-platform, as is the core `NetSparkleUpdater` library/class. Test on your own, of course, but there's no reason I can think of why such a scheme would not work with the right software permissions, at least in theory. In practice there may be other issues to consider on other OS's (e.g. sandboxing on macOS).

**Reply by balaji222528** [reply]
Thanks for the Reply. Is there any way to customize the interactive installer that runs on the MacOS ??As in with my own UI Look and Feel , logo etc ?

**Reply by Deadpikle** [reply]
`NetSparkleUpdater` does not manage installers, so I don't really have advice for you in that regard. You'll need to do some research on that on your own depending on your needs.

**Reply by balaji222528** [reply]
This is a different question. Is there any way I can customize the a--cast xml file ? as in adding my own element within the Item ? Please guide me here

**Reply by Deadpikle** [reply]
You can do whatever you like with the app cast file as long as the core elements that NetSparkle created already stay intact. If you are using signatures for your app cast file as well as your executables, don't forget to sign the updated app cast file using `netsparkle-generate-appcast --generate-signature path/to/appcast.xml` and  put that output in `appcast.signature`.

`NetSparkle` of course will not read those custom elements natively. You'll have to use your own `IAppCastHandler` that overrides the `ParseAppCast` function to pull out any information you're wanting, as well as making any other changes elsewhere in your code or via other class overrides as needed.

**Reply by balaji222528** [reply]
thanks is there a command to add a new item while generating the appcast file ?

**Reply by balaji222528** [reply]
Not sure how to run a command so that it sticks this new element in the item
